sdk/servicebus/azure-servicebus/api_samples.py

```diff
@@ -0,0 +1,219 @@
+    conn_str="conn_str",
+    queue_name="queue_name"
+)
+
+queue_receiver = ServiceBusReceiver(
+    fully_qualified_namespace="fully_qualified_namespace",
+    queue_name="queue_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session_id="test_session"
+)
+
+# Topic Sender
+topic_sender = ServiceBusSender.from_connection_string(
+    conn_str="conn_str",
+    topic_name="topic_name"
+)
+
+topic_sender = ServiceBusSender(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key")
+)
+
+# Subscription Receiver
+subscription_receiver = ServiceBusReceiver.from_connection_string(
+    conn_str="conn_str",
+    topic_name="topic_name",
+    subscription_name="subscription_name"
+)
+
+subscription_receiver = ServiceBusReceiver(
+    fully_qualified_namespace="fully_qualified_namespace",
+    topic_name="topic_name",
+    subscription_name="subcription_name",
+    credential=ServiceBusSharedKeyCredential("policy", "key"),
+    session_id="test_session"
+)
+
+
+### Send and receive
+# 1 Send
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+queue_sender = sb_client.get_queue_sender(queue_name="queue_name")
+
+with sb_client:
+    with queue_sender:
+        # 1.1 send single
+        message = Message("Test message")
+        queue_sender.send(message)
+
+        # 1.2 send batch
+        batch_message = topic_sender.create_batch(max_size_in_bytes=256 * 1204)
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        topic_sender.send(batch_message)
+
+        # 1.3 schedule
+        schedule_message = Message("scheduled_message")
+
+        schedule_time_utc = datetime.utcnow() + timedelta(minutes=10)
+        sequence_number = topic_sender.schedule(schedule_message, schedule_time_utc)
+
+        batch_schedule_message = topic_sender.create_batch(max_size_in_bytes=256 * 1204)
+
+        while True:
+            try:
+                batch_message.add(Message("Test message"))
+            except ValueError:
+                break
+
+        batch_sequence_numbers = topic_sender.schedule(batch_schedule_message, schedule_time_utc)
+
+        # 1.4 cancel schedule
+        topic_sender.cancel_scheduled_messages(sequence_number)
+        topic_sender.cancel_scheduled_messages(batch_sequence_numbers)
+
+# 2 Receive
+sb_client = ServiceBusClient.from_connection_string(conn_str="conn_str")
+queue_receiver = sb_client.get_queue_receiver(queue_name="queue_name")
+
+# 2.1 peek
+with sb_client:
+    with queue_receiver:
+        msgs = queue_receiver.peek(count=10)
+        for msg in msgs:
+            print(msg)
+
+# 2.2 iterator receive
+with sb_client:
+    with queue_receiver:
+        for msg in queue_receiver:
+            print(message)
+            msg.complete()
+            # msg.renew_lock()
```

Comment:
There's no auto-renewal in this model, correct?

Reply:
correct, it's not included.

Reply:
for msg.complete( ), does the message object has a reference to the connection/link object? If so this might not be good. A message can be passed down by the user to somewhere else. And the connection object might be already closed. This will mess up the connection lifecycle and prevent the connection object to be garbage collected.  
If message has no reference to connection/link/session or any other such objects, this has no problem.